lib/waterpig/database-cleaner.rb

```diff
@@ -25,18 +37,57 @@ def with_showing(show, detailed)
     end
   end
 
+  def cleaning_policy(config, metadata)
```

Comment:
complexity of this method is pretty high. Not worth holding up this merge for, but worth thinking about the architecture a bit  for down the line.